### PR TITLE
Client Notebook Auto Reload (#975)

### LINKED ISSUE #975: Auto reload page client side during notebook use
**Description**
An extension of #460.
We want the client side to also reload automatically given that the backend side knows when a file has been reloaded.

**Acceptance Criteria**
- [ ] Ensure new code is unit tested, and check code coverage is at least 90%
- [ ] Propagate any change on the demos and run all of them to ensure there is no breaking change
- [ ] Ensure any change is well documented

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Resolve #975

### CODE REVIEW

File: frontend/taipy-gui/src/components/Taipy/Navigate.tsx [hunk 1]
```diff
@@ -27,32 +27,45 @@ const Navigate = ({ to, params, tab, force }: NavigateProps) => {
...
-            let tprh: string | null = null;
-            let meta: string | null = null;
-            if (searchParams.has("tprh")) {
-                tprh = searchParams.get("tprh");
-                searchParams.delete("tprh");
-                if (searchParams.has("tp_cp_meta")) {
-                    meta = searchParams.get("tp_cp_meta");
-                    searchParams.delete("tp_cp_meta");
+            const filteredParams = params
+                ? Object.keys(params || {}).reduce((acc, key) => {
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Navigate.tsx]: is `|| {}` necessary if you test params before ?

File: frontend/taipy-gui/src/components/Taipy/Navigate.tsx [hunk 2]
```diff
@@ -27,32 +27,45 @@ const Navigate = ({ to, params, tab, force }: NavigateProps) => {
...
+                return navigate(0);
+            }
+            if (reloadSameRouteOnly) {
+                if (location.pathname === tos) {
+                    navigate(0);
                 }
+                return;
             }
+            // Handle Resource Handler Id
+            const tprh = params?.tprh;
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Navigate.tsx]: these could be in line 64 +

**Review by FredLL-Avaiga — DISMISSED** [review]

File: taipy/gui/_renderers/utils.py [hunk 3]
```diff
@@ -140,5 +140,5 @@ def on_modified(self, event):
         if Path(event.src_path).resolve() == Path(self._file_path).resolve():
             self._renderer.set_content(self._file_path)
             _TaipyLogger._get_logger().info(
-                f"File '{self._file_path}' has been modified. Reload your page to see the changes."
+                f"File '{self._file_path}' has been modified. Page will be reloaded automatically."
```
Comment (FabienLelaquais) [inline taipy/gui/_renderers/utils.py]:
```suggestion
                f"File '{self._file_path}' has been modified."
```

**Review by FabienLelaquais — DISMISSED** [review]
😶

File: frontend/taipy-gui/src/components/Taipy/Navigate.tsx (hunk 1 above)
Comment (dinhlongviolin1) [reply frontend/taipy-gui/src/components/Taipy/Navigate.tsx]: you are right, it is kinda redundant here

**Review by FabienLelaquais — APPROVED** [review]
👍

**Review by FredLL-Avaiga — APPROVED** [review]

**Pull request merged**